cpp/src/toppra/solver/seidel-parallel.cpp

```diff
@@ -0,0 +1,402 @@
+#include <toppra/solver/seidel-parallel.hpp>
+#include <toppra/solver/seidel-internal.hpp>
+
+namespace toppra {
+namespace solver {
+
+namespace seidel {
+const LpSol INFEASIBLE { false };
+
+/// Compute value coeffs * [v 1] of a constraint.
+/// handling infinite values for v as, when v[i] is +/- infinity,
+/// c[i]*v[i] == 0 if v[i] == 0.
+template<class Coeffs, class Vars>
+typename Coeffs::Scalar value(const Eigen::MatrixBase<Coeffs>& coeffs,
+    const Eigen::MatrixBase<Vars>& vars)
+{
+  static_assert(Coeffs::RowsAtCompileTime == 1
+      && Vars::ColsAtCompileTime == 1
+      && Coeffs::ColsAtCompileTime == Vars::RowsAtCompileTime + 1,
+      "Size mismatch between coefficient (1x(N+1)) and vars (Nx1)");
+  typename Coeffs::Scalar res = coeffs(coeffs.cols()-1);
+  for (int i = 0; i < Vars::RowsAtCompileTime; ++i)
+    res += (coeffs[i]==0 && !std::isfinite(vars[i]) ? 0 : coeffs[i]*vars[i]);
+  return res;
+}
+
+constexpr value_type infi = 1e6;
```

Comment:
This *infinity* is definitely too low and I don't think this is actually necessary. What issue did you get that made you set this ?

Reply:
If I set infi to a very large number, it will cause some numerical calculation problems. It may be caused by the limitations of the algorithm itself. In fact, the original seidel algorithm also set an upper limit of 100, which is much smaller than the upper limit here.

Reply:
In the C++ implementation, I don't think there is any such limit. Are you talking about the limit discussed in #156 ?
There is no good reason for this limit and this limit is modifiable by the user since #160.
Through this API, I change the bound to [ 0, +inf ] in my code and it works well on my use cases.

Reply:
You are right. In the origin develop branch, if you set m_initXBound to [ 0, +inf ], it will works well. But it doesn't work in parallel branch here. This is also the limitation of this parallel algorithm.
There are two variables, u and x. In principle, the limit of x (which is m_initXBound in origin develop branch) should be set to [ 0, +inf ], and the limit of u should be set to [ -inf, +inf ]. 
However, after a series of incomplete tests, I found that in parallel branch here, the limit of x can be set to [ 0, +inf ], but the limit of u should be set to [ -1e13, +1e13 ] or smaller. Otherwise, this parallel algorithm will have a large numerical error. 
The reason is that the order of constraints in the LP in this parallel algorithm has been adjusted. In order to process unrelated constraints in parallel, I put the two constraints corresponding to xNext at the end. This leads to a high degree of linear correlation between the constraints during the first and last stages of parallel processing, and the difference between the variables u and x is even up to 15 orders of magnitude. This is enough to make the "double“ data structure produce enough errors to affect the result.
So I set infi to 1e6, it can also be set larger, of course, a more complete test is needed. A smaller infi setting may speed up the calculation speed, and it is also sufficient to meet actual data requirements.